### PR TITLE
NODE-315 Matcher invalid\expired orders cleanup routine

### PR DESCRIPTION
**Pull request by ebceu4**

### CODE REVIEW

File: src/main/scala/com/wavesplatform/matcher/market/OrderBookActor.scala [hunk 1]
```diff
@@ -28,6 +28,8 @@ import scala.concurrent.duration._
 
 import com.wavesplatform.network._
 
+case class OrderCleanup()
```
Comment (ismagin) [inline src/main/scala/com/wavesplatform/matcher/market/OrderBookActor.scala]: case object

File: src/main/scala/com/wavesplatform/matcher/market/OrderHistoryActor.scala [hunk 2]
```diff
@@ -114,6 +116,12 @@ class OrderHistoryActor(val settings: MatcherSettings, val utxPool: UtxPool, val
     }
   }
 
+  def ordersCleanup(orderBook: OrderBook, ts: Long): Either[GenericError, Seq[String]] = {
+    val result = orderBook.asks.values.++(orderBook.bids.values).flatten.filterNot(x => x.order.isValid(ts)).map(_.order.idStr).toSeq
```
Comment (koloale) [inline src/main/scala/com/wavesplatform/matcher/market/OrderHistoryActor.scala]: Use validateNewOrder here

File: src/main/scala/com/wavesplatform/matcher/market/OrderBookActor.scala [hunk 3]
```diff
@@ -60,6 +62,8 @@ class OrderBookActor(assetPair: AssetPair,
       onAddOrder(order)
     case cancel: CancelOrder =>
       onCancelOrder(cancel)
+    case OrderCleanup  =>
```
Comment (koloale) [inline src/main/scala/com/wavesplatform/matcher/market/OrderBookActor.scala]: Where is this message sent? Should be scheduled in a configurable period of time (default 5 min)

File: src/main/scala/com/wavesplatform/matcher/market/OrderBookActor.scala [hunk 4]
```diff
@@ -110,16 +117,23 @@ class OrderBookActor(assetPair: AssetPair,
     context.become(waitingValidation)
   }
 
-  def handleValidateCancelResult(res: Either[GenericError, CancelOrder]): Unit = {
+  def onOrderCleanup(orderBook: OrderBook): Unit = {
+    orderHistory ! OrdersCleanup(orderBook, NTP.correctedTime())
+    apiSender = None
+    cancellable = Some(context.system.scheduler.scheduleOnce(ValidationTimeout, self, ValidationTimeoutExceeded))
```
Comment (koloale) [inline src/main/scala/com/wavesplatform/matcher/market/OrderBookActor.scala]: No need to handle timeout and become readonly

**Pull request merged**